storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/V4FinalizedKvStoreDao.java

```diff
@@ -116,6 +122,54 @@ public void close() throws Exception {
     return db.get(schema.getColumnNonCanonicalBlocksByRoot(), root);
   }
 
+  @Override
+  public void ingest(
+      final KvStoreFinalizedDao finalizedDao, final int batchSize, final Consumer<String> logger) {
+    Preconditions.checkArgument(batchSize > 1, "Batch size must be greater than 1 element");
+    Preconditions.checkArgument(finalizedDao instanceof V4FinalizedKvStoreDao);
```

Comment:
Probably worth providing an error string here to explain the problem - by default it will basically just say "Precondition failed".